native/src/modules/app/components/App.js

```diff
@@ -45,22 +46,22 @@ const store: Store<StateType, StoreActionType> = createReduxStore(dataContainer)
 const App = () => {
   const [routeName, setRouteName] = useState<?string>(null)
   const [routeKey, setRouteKey] = useState<?string>(null)
-  const [cityCode, setCityCode] = useState<?string>(null)
-  const [languageCode, setLanguageCode] = useState<?string>(null)
+  const [routeIndex, setRouteIndex] = useState<number>(0)
 
-  const onStateChange = useCallback(state => {
-    if (state) {
-      const route = state.routes[state.index]
-      setRouteName(route.name)
-      setRouteKey(route.key)
-      if (route.params?.cityCode) {
-        setCityCode(typeof route.params.cityCode === 'string' ? route.params.cityCode : null)
+  const onStateChange = useCallback(
+    state => {
+      if (state) {
+        if (state.index === routeIndex - 1) {
```

Comment:
Is there a reason, why you decided to work with index instead of route name.

In the docu it is done with the route name and i am not sure it the index is less robust and there may be less suitable.